contracts/YieldStEthLever.sol

```diff
@@ -66,279 +75,391 @@ interface YieldLadle {
+    /// @notice The Yield Cauldron, handles debt and collateral balances.
     ICauldron constant cauldron =
         ICauldron(0xc88191F8cb8e6D4a668B047c1C8503432c3Ca867);
-    IStableSwap stableSwap =
+    /// @notice Curve.fi token swapping contract between Ether and StETH.
+    IStableSwap constant stableSwap =
         IStableSwap(0x828b154032950C8ff7CF8085D841723Db2696056);
-    StEthConverter constant stEthConverter =
-        StEthConverter(0x93D232213cCA6e5e7105199ABD8590293C3eb106);
-    bytes6 constant ilkId = bytes6(0x303400000000); //wsteth
+    /// @notice The ild ID for WStEth.
+    bytes6 constant ilkId = bytes6(0x303400000000);
+    /// @notice The Yield Protocol Join containing WstEth.
     FlashJoin constant flashJoin =
         FlashJoin(0x5364d336c2d2391717bD366b29B6F351842D7F82);
-    IPool pool = IPool(0xc3348D8449d13C364479B1F114bcf5B73DFc0dc6);
-    Giver giver;
-    address constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
-    address constant wsteth = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
-    address constant steth = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84;
+    /// @notice The Yield Protocol Join containing Weth.
     FlashJoin constant flashJoin2 =
-        FlashJoin(0x3bDb887Dc46ec0E964Df89fFE2980db0121f0fD0); //WETH JOIN
-
-    constructor(IERC3156FlashLender fyToken_, Giver giver_) {
+        FlashJoin(0x3bDb887Dc46ec0E964Df89fFE2980db0121f0fD0);
+    /// @notice Ether Yield liquidity pool. Exchanges Weth with FYWeth.
+    IPool constant pool = IPool(0xc3348D8449d13C364479B1F114bcf5B73DFc0dc6);
+    /// @notice FyWeth, used to borrow based on Weth.
+    FYToken constant fyToken = FYToken(0x53358d088d835399F1E97D2a01d79fC925c7D999);
+    /// @notice The Giver contract can give vaults on behalf on a user who gave
+    ///     permission.
+    Giver immutable giver;
+
+    /// @notice Deploy this contract.
+    /// @param giver_ The `Giver` contract to use.
+    /// @dev The contreact should never own anything in between transactions;
+    ///     no tokens, no vaults. To save gas we give these tokens full
+    ///     approval.
+    constructor(Giver giver_) {
         giver = giver_;
-        fyToken = fyToken_;
-
-        IERC20(address(fyToken_)).approve(address(pool), type(uint256).max);
-        pool.base().approve(address(stableSwap), type(uint256).max);
-        IERC20(wsteth).approve(address(stableSwap), type(uint256).max); //wsteth
-        IERC20(steth).approve(address(stableSwap), type(uint256).max); //steth
-        IERC20(weth).approve(
-            0x3bDb887Dc46ec0E964Df89fFE2980db0121f0fD0, //join
-            type(uint256).max
-        ); //weth
-        IERC20(wsteth).approve(
-            0x5364d336c2d2391717bD366b29B6F351842D7F82, //flashjoin
-            type(uint256).max
-        ); //wsteth
+
+        // TODO: What if these approvals fail by returning `false`? Is that even a case worth
+        //  considering?
```

Comment:
There is a `safeApprove` method somewhere that we can add to our `TransferHelper`. No reason not to use it.

Reply:
I have [opened a PR](https://github.com/yieldprotocol/yield-utils-v2/pull/44) to add two methods to increase/decrease allowance, based on code by OpenZeppelin. They also have a `safeApprove` but it is deprecated because it has many of the same issues as `approve`.